src/widget/tooltipqopengl.cpp

```diff
@@ -0,0 +1,46 @@
+#include "widget/tooltipqopengl.h"
+
+#include <widget/wglwidget.h>
+
+#include <QStyle>
+#include <QTimer>
+#include <QToolTip>
+#include <memory>
+
+ToolTipQOpenGL::ToolTipQOpenGL() {
+    m_timer.setSingleShot(true);
+    connect(&m_timer, &QTimer::timeout, this, &ToolTipQOpenGL::onTimeout);
+}
+
+void ToolTipQOpenGL::onTimeout() {
+    if (m_widget) {
+        QToolTip::showText(m_pos, m_widget->toolTip(), m_widget);
+    }
+}
+
+ToolTipQOpenGL::~ToolTipQOpenGL() {
+}
```

Comment:
no need for default ctor, especially if empty.

Reply:
True that. I'll remove it.